src/protocols/user_directed_commissioning/UserDirectedCommissioningClient.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file implements an object for a CHIP Echo unsolicitied
+ *      initiator (client).
+ *
+ */
+
+#include "UserDirectedCommissioning.h"
+
+namespace chip {
+namespace Protocols {
+namespace UserDirectedCommissioning {
+
+CHIP_ERROR UserDirectedCommissioningClient::SendUDCMessage(TransportMgrBase * transportMgr, System::PacketBufferHandle && payload,
+                                                           chip::Inet::IPAddress commissioner, uint16_t port)
+{
+    CHIP_ERROR err;
+
+    PayloadHeader payloadHeader;
+    PacketHeader packetHeader;
+
+    payloadHeader.SetMessageType(Protocols::UserDirectedCommissioning::Id, (uint8_t) MsgType::IdentificationDeclaration)
+        .SetInitiator(true);
+    payloadHeader.SetNeedsAck(false);
+
+    VerifyOrReturnError(!payload.IsNull(), CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrReturnError(!payload->HasChainedBuffer(), CHIP_ERROR_INVALID_MESSAGE_LENGTH);
+    VerifyOrReturnError(payload->TotalLength() <= kMaxAppMessageLen, CHIP_ERROR_MESSAGE_TOO_LONG);
+
+    ReturnErrorOnFailure(payloadHeader.EncodeBeforeData(payload));
+
+    packetHeader.SetEncryptionType(Header::EncryptionType::kEncryptionTypeNone);
+
+    uint16_t totalLen = payload->TotalLength();
+    uint16_t taglen   = 0;
+    ChipLogDetail(Inet, "SendUDCMessage totalLen=%d taglen=%d", totalLen, taglen);
+
+    payload->SetDataLength(static_cast<uint16_t>(totalLen + taglen));
```

Comment:
This line makes no sense.  Given our above checks, the data length is already that value.  What is this code trying to accomplish?

Reply:
addressed in latest changes